djstripe/contrib/rest_framework/urls.py

```diff
@@ -23,6 +23,22 @@
 app_name = "djstripe_rest_framework"
 
 urlpatterns = [
-    # REST api
-    path("subscription/", views.SubscriptionRestView.as_view(), name="subscription")
```

Comment:
removing this is a breaking change, can we leave it in place?  Maybe with a deprecation warning (warning.warn) that it will be removed in a future version?

Reply:
There are various dimensions in your question. 

- Removing the route (as it is now) will return "404" not found. One possibility would be to reintroduce the route with a small function to return a "400" bad request, and a message saying: "use the new routes"... This is the simplest, but maybe too abrupt.

- I can also reintroduce the route, and the old view class, and put somewhere a warning. But the warning will appear only in server logs. It will have no effect on the REST requests. Not really helpful, isn't it?

- The consequence of the point above is then to have 3 routes with a behavior which is not entirely consistent. I see two main inconsistencies. First, the `GET` request without any parameter on the old route was providing the details of the subscription of the authenticated user, which is "wrong" in the REST sense, but maybe temporarily acceptable. (for the record, now you have to use the detail endpoint and provide the correct id.)

The second inconsistency is the replacement of the `delete` method with a `put` since it has the effect of updating the resource and not deleting it in the REST sense (the subscription is indeed cancelled, but the resource is still here).

- Thus the last solution would be to reintroduce the route, and return "redirects" response (301 or better 302 permanent redirect). This is fairly easy for a `GET` request as it involves no payload. A bit more complicated for a POST request with a payload. And the DELETE meaning will be changed anyway...

What do you prefer ?

Reply:
> 
> 
> There are various dimensions in your question.
> 
>     * Removing the route (as it is now) will return "404" not found. One possibility would be to reintroduce the route with a small function to return a "400" bad request, and a message saying: "use the new routes"... This is the simplest, but maybe too abrupt.
> 
>     * I can also reintroduce the route, and the old view class, and put somewhere a warning. But the warning will appear only in server logs. It will have no effect on the REST requests. Not really helpful, isn't it?

Yeah I'd say leave the old route + view, add a deprecation warning, and remove in the next minor release (2.x+1).

*hopefully* people are looking at their logs for deprecation warnings when they upgrade packages and/or run tests?  🤞 

If they're not there's not much we can do to help them hah.

>     * The consequence of the point above is then to have 3 routes with a behavior which is not entirely consistent. I see two main inconsistencies. First, the `GET` request without any parameter on the old route was providing the details of the subscription of the authenticated user, which is "wrong" in the REST sense, but maybe temporarily acceptable. (for the record, now you have to use the detail endpoint and provide the correct id.)
> 
> 
> The second inconsistency is the replacement of the `delete` method with a `put` since it has the effect of updating the resource and not deleting it in the REST sense (the subscription is indeed cancelled, but the resource is still here).

Yeah, tricky one - For what it's worth stripe uses DELETE for this (see https://stripe.com/docs/api/subscriptions/cancel ), and their API is generally pretty well designed, my gut says it's better to stick close to them.  What do you think @kavdev ?

> 
>     * Thus the last solution would be to reintroduce the route, and return "redirects" response (301 or better 302 permanent redirect). This is fairly easy for a `GET` request as it involves no payload. A bit more complicated for a POST request with a payload. And the DELETE meaning will be changed anyway...

Yeah like I say I'd say just leave the old route for now, just add a deprecation warning (and note it in the change log).

Reply:
Ok. As for the `delete` method in the new Detail View, the only solution I find for now is to implement it (this will allow a DELETE request on that view), but I override the default behavior (which is to destroy the object) to actually simulate an update which cancels the Subscription.

Reply:
Yep, that can be the same as the existing view's delete I think?

https://github.com/dj-stripe/dj-stripe/blob/ba06ce4323e277a6362d6086e185182a386d6019/djstripe/contrib/rest_framework/views.py#L67-L84

Reply:
Yes